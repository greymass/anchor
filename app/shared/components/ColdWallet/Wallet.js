// @flow
import React, { Component } from 'react';
import { Grid } from 'semantic-ui-react';

import ColdWalletInfo from './Wallet/Info';
import ColdWalletPanel from './Wallet/Panel';
import ColdWalletTransaction from './Wallet/Transaction';

type Props = {
  actions: {},
  keys: {},
  settings: {},
  transaction: {},
  wallet: {},
  system: {}
};

export default class ColdWallet extends Component<Props> {
  props: Props;

  render() {
    const {
      actions,
      keys,
      settings,
      system,
      transaction,
      wallet
    } = this.props;
    return (
      <Grid divided>
        <Grid.Row>
          <Grid.Column width={6}>
            <ColdWalletPanel
              actions={actions}
              keys={keys}
              settings={settings}
              system={system}
              wallet={wallet}
            />
          </Grid.Column>
          <Grid.Column width={10}>
            {(transaction.data)
              ? (
                <ColdWalletTransaction
                  actions={actions}
                  keys={keys}
                  transaction={transaction}
                />
              )
              : (
                <ColdWalletInfo />
              )
            }
          </Grid.Column>
        </Grid.Row>
      </Grid>
    );
  }
}
