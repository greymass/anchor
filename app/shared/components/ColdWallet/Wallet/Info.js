// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';

import { Divider, Icon, Header, List, Segment } from 'semantic-ui-react';

class ColdWalletInfo extends Component<Props> {
  render() {
    const { t } = this.props;
    return (
      <Segment basic padded style={{ paddingTop: 0 }}>
        <Segment attached="top" secondary textAlign="center">
          <Header icon size="large">
            <Icon
              color="blue"
              name="snowflake"
            />
            {t('coldwallet_wallet_title')}
            <Header.Subheader
              content={t('coldwallet_wallet_description')}
            />
          </Header>
        </Segment>
        <Segment attached>
          <Header>
            {t('coldwallet_wallet_instructions_title')}
          </Header>
          <p>{t('coldwallet_wallet_instructions_1')}</p>
          <Divider />
          <Segment basic padded>
            <Header>
              {t('coldwallet_wallet_workflow')}
            </Header>
            <List ordered relaxed>
              <List.Item>
                {t('coldwallet_wallet_instructions_2')}
              </List.Item>
              <List.Item>
                {t('coldwallet_wallet_instructions_3')}
              </List.Item>
              <List.Item>
                {t('coldwallet_wallet_instructions_4')}
              </List.Item>
              <List.Item>
                {t('coldwallet_wallet_instructions_5')}
              </List.Item>
              <List.Item>
                {t('coldwallet_wallet_instructions_6')}
              </List.Item>
              <List.Item>
                {t('coldwallet_wallet_instructions_7')}
              </List.Item>
              <List.Item>
                {t('coldwallet_wallet_instructions_8')}
              </List.Item>
              <List.Item>
                {t('coldwallet_wallet_instructions_9')}
              </List.Item>
              <List.Item>
                {t('coldwallet_wallet_instructions_10')}
              </List.Item>
            </List>

          </Segment>
        </Segment>
      </Segment>
    );
  }
}

export default translate('coldwallet')(ColdWalletInfo);
