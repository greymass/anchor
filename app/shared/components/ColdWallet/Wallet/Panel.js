// @flow
import React, { Component } from 'react';

import WalletPanelLocked from '../../Wallet/Panel/Locked';
import ColdWalletPanelUnlocked from './Panel/Unlocked';

type Props = {
  actions: {},
  settings: {},
  system: {},
  keys: {},
  validate: {},
  wallet: {}
};

export default class ColdWalletPanel extends Component<Props> {
  render() {
    const {
      actions,
      keys,
      settings,
      system,
      validate,
      wallet
    } = this.props;

    let panel = false;
    if ((!keys || !keys.key) && wallet.data) {
      panel = (
        <WalletPanelLocked
          actions={actions}
          settings={settings}
          validate={validate}
          wallet={wallet}
        />
      );
    }
    if (keys && keys.key) {
      panel = (
        <ColdWalletPanelUnlocked
          actions={actions}
          keys={keys}
          settings={settings}
          system={system}
        />
      );
    }
    return (
      <div>
        {panel}
      </div>
    );
  }
}
