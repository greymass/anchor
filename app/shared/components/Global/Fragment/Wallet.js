// @flow
import React, { PureComponent } from 'react';
import { Header } from 'semantic-ui-react';
import { withTranslation } from 'react-i18next';
import Blockies from 'react-blockies';

export class GlobalFragmentWallet extends PureComponent<Props> {
  render() {
    const {
      account,
      address,
      authorization,
      authAccount,
      authAuthorization,
      disableAvatar,
      mode,
      size,
    } = this.props;
    return (
      <Header
        size={size || 'tiny'}
        style={{
          margin: 0
        }}
      >
        {(!disableAvatar)
          ? (
            <Blockies
              className="ui image"
              seed={`${account}@${authorization}`}
            />
          )
          : false
        }
        <Header.Content style={{ minWidth: (disableAvatar) ? 'none' : '10em' }}>
          <React.Fragment>
            {address || account}
            <Header.Subheader>
              <React.Fragment>
                {(address) ? `${account} / ` : false}
                {authorization}
                {' / '}
                {(mode === 'auth') ? `${authAccount}@${authAuthorization}` : mode}
              </React.Fragment>
            </Header.Subheader>
          </React.Fragment>
        </Header.Content>
      </Header>
    );
  }
}

export default withTranslation('global')(GlobalFragmentWallet);
