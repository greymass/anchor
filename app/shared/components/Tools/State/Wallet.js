// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';

import { Button, Header, Grid, Message, Segment } from 'semantic-ui-react';
import ReactJson from 'react-json-view';

class ToolsStateWallet extends Component<Props> {
  render() {
    const {
      settings,
      t
    } = this.props;
    return (
      <Segment basic>
        <Header>
          {t('tools_wallet_state_header')}
          <Header.Subheader>
            {t('tools_wallet_state_subheader')}
          </Header.Subheader>
        </Header>
        <Segment basic>
          <ReactJson
            displayDataTypes={false}
            displayObjectSize={false}
            iconStyle="square"
            name={null}
            src={settings}
            style={{ padding: '1em' }}
            theme="harmonic"
          />
        </Segment>
      </Segment>
    );
  }
}

export default translate('tools')(ToolsStateWallet);
