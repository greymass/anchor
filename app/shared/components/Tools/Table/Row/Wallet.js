// @flow
import React, { Component } from 'react';
import { withTranslation } from 'react-i18next';
import { find, findIndex } from 'lodash';

import { Button, Dropdown, Header, Icon, Label, Popup, Table } from 'semantic-ui-react';

import GlobalAccountEdit from '../../../../containers/Global/Account/Edit';
import GlobalButtonElevate from '../../../../containers/Global/Button/Elevate';
import GlobalFragmentAuthorization from '../../../Global/Fragment/Authorization';
import GlobalFragmentChainLogo from '../../../Global/Fragment/ChainLogo';
import GlobalFragmentWalletType from '../../../Global/Fragment/WalletType';
import GlobalButtonWalletUpgrade from '../../../../containers/Global/Button/Wallet/Upgrade';
import GlobalAccountConvertLedger from '../../../../containers/Global/Account/Convert/Ledger';
import EOSAccount from '../../../../utils/EOS/Account';
import WalletLockState from '../../../Wallet/LockState';

const initialLedgerState = {
  convertToLedgerAccount: undefined,
  convertToLedgerAuthorization: undefined,
  convertToLedgerKey: undefined,
};

const initialEditState = {
  editAccount: undefined,
  editAuthorization: undefined,
};

class ToolsTableRowWallet extends Component<Props> {
  constructor(props) {
    super(props);
    this.state = Object.assign(
      {},
      initialEditState,
      initialLedgerState,
    );
  }
  editWallet = (account, authorization) => {
    this.setState({
      editAccount: account,
      editAuthorization: authorization,
    });
  };
  removeWallet = (account, authorization) => {
    const { actions, settings, wallets } = this.props;

    actions.removeWallet(settings.chainId, account, authorization);

    if (wallets.length === 1) {
      return actions.changeModule('');
    }

    if (settings.account === account) {
      const otherWallet = wallets.find(wallet => (
        wallet.account !== account
        && wallet.authorization !== authorization
      ));
      if (otherWallet) {
        actions.useWallet(settings.chainId, otherWallet.account, otherWallet.authorization);
      }
    }
  };
  swapWallet = (account, authorization, password = false) => {
    const { actions, settings } = this.props;
    actions.useWallet(settings.chainId, account, authorization);
    if (password) {
      actions.unlockWallet(password);
    }
  };
  convertToLedger = (account, authorization, key = undefined) => this.setState({
    convertToLedgerAccount: account,
    convertToLedgerAuthorization: authorization,
    convertToLedgerKey: key,
  });
  resetLedgerConversion = () => this.setState(Object.assign(
    this.state,
    initialLedgerState
  ));
  resetEditWallet = () => this.setState(Object.assign(
    this.state,
    initialEditState
  ));
  render() {
    const {
      current,
      blockchains,
      pubkeys,
      settings,
      status,
      t,
      validate,
      wallet,
    } = this.props;
    const {
      account,
      authorization,
      address,
      chainId,
      mode,
      pubkey
    } = wallet;
    const {
      accountData
    } = current;
    const blockchain = find(blockchains, { chainId });
    const data = new EOSAccount(accountData).getPermission(authorization);
    const {
      convertToLedgerAccount,
      convertToLedgerAuthorization,
      convertToLedgerKey,
      editAccount,
      editAuthorization,
    } = this.state;
    let modal;
    let color = 'grey';
    if (convertToLedgerAccount && convertToLedgerAuthorization) {
      modal = (
        <GlobalAccountConvertLedger
          account={convertToLedgerAccount}
          authorization={convertToLedgerAuthorization}
          data={data}
          pubkey={convertToLedgerKey}
          onClose={this.resetLedgerConversion}
        />
      );
    }
    if (editAccount && editAuthorization) {
      modal = (
        <GlobalAccountEdit
          account={editAccount}
          authorization={editAuthorization}
          data={wallet}
          onClose={this.resetEditWallet}
        />
      );
    }
    const items = [
      (
        <Dropdown.Header key="advanced-header" icon="warning sign" content={t('wallet:wallet_advanced_header')} />
      ),
      (
        <Dropdown.Item
          content={t('wallet:view')}
          icon="edit"
          key="edit"
          onClick={() => this.editWallet(account, authorization)}
        />
      )
    ];
    // Is this the current wallet? Account + Authorization must match
    const isCurrentWallet = (
      account === current.account
      && authorization === current.authorization
    );
    // Is this either a Hot/Watch wallet and is there
    //   a Ledger currently connected? If so, append convert
    if (isCurrentWallet && ['hot', 'watch'].indexOf(mode) >= 0 && status === 'connected') {
      items.push((
        <Dropdown.Item
          content={t('wallet:wallet_convert_to_ledger')}
          // disabled={isCurrentWallet}
          icon="usb"
          key="ledger"
          onClick={() => this.convertToLedger(account, authorization, pubkey)}
        />
      ));
    }
    let icon = 'disk';
    // Create delete button based on wallet
    switch (mode) {
      case 'ledger': {
        color = 'purple';
        icon = 'usb';
        items.push((
          <Dropdown.Item
            content={t('wallet:wallet_remove')}
            icon="trash"
            key="delete"
            onClick={() => this.removeWallet(account, authorization)}
          />
        ));
        break;
      }
      case 'watch': {
        color = 'grey';
        icon = 'eye';
        items.push((
          <Dropdown.Item
            content={t('wallet:wallet_remove')}
            icon="trash"
            key="delete"
            onClick={() => this.removeWallet(account, authorization)}
          />
        ));
        break;
      }
      case 'cold': {
        color = 'blue';
        icon = 'snowflake';
        items.push((
          <Dropdown.Item
            content={t('wallet:wallet_remove')}
            icon="trash"
            key="delete"
            onClick={() => this.removeWallet(account, authorization)}
          />
        ));
        break;
      }
      default: {
        color = 'green';
        icon = 'id card';
        items.push((
          <GlobalButtonElevate
            key="remove"
            onSuccess={() => this.removeWallet(account, authorization)}
            settings={settings}
            trigger={(
              <Dropdown.Item
                icon="trash"
                key="delete"
                text={t('wallet:wallet_remove')}
              />
            )}
            validate={validate}
            wallet={wallet}
          />
        ));
      }
    }
    const unlocked = (pubkeys.unlocked.includes(pubkey));
    return (
      <Table.Row key={`${account}-${authorization}`}>
        <Table.Cell collapsing>
          {modal}
          <Header size="small">
            <GlobalFragmentAuthorization
              account={account}
              address={address}
              authorization={authorization}
              pubkey={pubkey}
            />
          </Header>
        </Table.Cell>
        <Table.Cell collapsing textAlign="center">
          <GlobalFragmentChainLogo
            avatar
            chainId={blockchain.chainId}
            name={blockchain.name}
          />
        </Table.Cell>
        <Table.Cell>
          <GlobalFragmentWalletType
            mode={mode}
          />
        </Table.Cell>
        <Table.Cell collapsing textAlign="right">
          {(!unlocked && (mode === 'hot' || mode === 'cold' || mode === 'auth'))
            ? (
              <GlobalButtonElevate
                onSuccess={(password) => this.swapWallet(account, authorization, password)}
                settings={settings}
                trigger={(
                  <Button
                    color="green"
                    content={t('tools_wallets_swap')}
                    disabled={isCurrentWallet}
                    icon="random"
                  />
                )}
                validate={validate}
                wallet={wallet}
              />
            )
            : false
          }
          {(mode === 'watch' || mode === 'ledger' || unlocked)
            ? (
              <Button
                color="green"
                content={t('tools_wallets_swap')}
                disabled={isCurrentWallet}
                icon="random"
                onClick={() => this.swapWallet(account, authorization)}
              />
            )
            : false
          }
          <Dropdown
            direction="left"
            floating
            button
            className="icon"
            icon="ellipsis vertical"
          >
            <Dropdown.Menu>
              {items}
            </Dropdown.Menu>
          </Dropdown>
        </Table.Cell>
      </Table.Row>
    );
  }
}

export default withTranslation('tools')(ToolsTableRowWallet);
