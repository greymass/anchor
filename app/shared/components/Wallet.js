// @flow
import React, { Component } from 'react';
import { Grid } from 'semantic-ui-react';

import SidebarConnection from '../containers/Sidebar/Connection';

import WalletPanel from './Wallet/Panel';
import WalletStatus from './Wallet/Status';

type Props = {
  actions: {},
  actionObjects: {},
  accounts: {},
  globals: {},
  keys: {},
  settings: {},
  validate: {},
  wallet: {},
  balances: {},
  system: {},
  transaction: {}
};

export default class Wallet extends Component<Props> {
  props: Props;

  render() {
    const {
      actions,
      actionObjects,
      accounts,
      balances,
      globals,
      keys,
      settings,
      system,
      transaction,
      validate,
      wallet
    } = this.props;

    return (
      <Grid divided>
        <Grid.Row>
          <Grid.Column width={6}>
            <SidebarConnection />
            <WalletPanel
              actions={actions}
              accounts={accounts}
              balances={balances}
              keys={keys}
              settings={settings}
              system={system}
              transaction={transaction}
              validate={validate}
              wallet={wallet}
            />
          </Grid.Column>
          <Grid.Column width={10}>
            <WalletStatus
              actions={actions}
              actionObjects={actionObjects}
              accounts={accounts}
              balances={balances}
              globals={globals}
              keys={keys}
              settings={settings}
              validate={validate}
              wallet={wallet}
            />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    );
  }
}
