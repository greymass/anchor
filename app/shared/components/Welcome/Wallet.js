// @flow
import React, { Component } from 'react';
import { Header, Segment } from 'semantic-ui-react';
import { translate } from 'react-i18next';
import WelcomeWalletContainer from '../../containers/Welcome/Wallet';

class WelcomeWallet extends Component<Props> {
  render() {
    const {
      onStageSelect,
      t
    } = this.props;
    return (
      <Segment
        size="small"
        stacked
      >
        <Header>
          {t('welcome_stage')} #4: {t('welcome_stage_wallet')}
        </Header>
        <p>{t('welcome_instructions_6')}</p>
        <WelcomeWalletContainer
          editing
          onStageSelect={onStageSelect}
        />
      </Segment>
    );
  }
}

export default translate('welcome')(WelcomeWallet);
