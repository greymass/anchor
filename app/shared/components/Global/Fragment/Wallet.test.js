import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalFragmentWallet } from './Wallet';

const mockProps = {
  account: {},
  authorization: {},
  mode: 'hot',
  pubkey: 'EOSXXX',
};

describe('GlobalFragmentWallet', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalFragmentWallet {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
