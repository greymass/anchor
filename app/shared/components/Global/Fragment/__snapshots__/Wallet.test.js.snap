// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalFragmentWallet renders correctly 1`] = `
<Header
  size="tiny"
  style={
    Object {
      "margin": 0,
    }
  }
>
  <Identicon
    className="ui image"
    seed="[object Object]@[object Object]"
  />
  <HeaderContent
    style={
      Object {
        "minWidth": "10em",
      }
    }
  >
    <Component />
    <HeaderSubheader>
      <Component />
       / 
      hot
    </HeaderSubheader>
  </HeaderContent>
</Header>
`;
